pkgs/applications/misc/st/xst.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchurl, pkgconfig, libX11, ncurses, libXext, libXft , fontconfig
+, patches ? [], extraLibs ? []}:
+
+with stdenv.lib;
+
+let
+  version = "0.7.1";
+  name = "xst-${version}";
+  patches' = if patches == null then [] else patches;
```

Comment:
```
nix-repl> let a = {foo ? []}: foo;
          in a { foo = null; }
null
nix-repl> let a = {foo ? []}: foo;
          in a { foo = [1 2 3]; }
[ 1 2 3 ]
```

it will assign a default value you see here.